flow2ml/Flow.py

```diff
@@ -37,8 +39,64 @@ def __init__(self,dataset_dir,data_dir):
       self.classes.remove('.ipynb_checkpoints') 
     except:
       pass
+    
+    self.results_path = os.path.join(os.getcwd(),"GeneratedReports")
+
+    try:
+      os.mkdir(self.results_path)
+    except:
+      raise Exception("Unable to create directory for results.")
 
     self.num_classes =  len(self.classes)
+    self.class_counts = {}
+    self.categoriesCountPlot()
+
+  def categoriesCountPlot(self):
+  
+    '''
+      Function used to create a countPlot of all categories of images
+      Args : None
+    '''
+
+    # loop through the folders of each class, store the class and number of images in dictionary
+    for folder in self.classes:
+      path = os.path.join(self.dataset_dir ,self.data_dir, folder)
+      count = len(os.listdir(path))
+      self.class_counts[folder] = count
+    
+    # display the count on the console
+    print("Counts per each category")
+    for key in self.class_counts.keys():
+      print(f"Category {key}: {self.class_counts[key]}")
+
+    # check for class imbalance
+    min_class = min(self.class_counts.values())
+    max_class = max(self.class_counts.values())
+    total_images = sum(self.class_counts.values())
+    if max_class - min_class >= 0.5 * total_images:
```

Comment:
Can you please add range here from 5% to 15%

Reply:
It's 50%, should I change it to range of 5% to 15%?